### ISSUE TITLE
"Night" feature: NPCs don't update until interaction by player

### ISSUE THREAD

**Issue by Pilzinsel64** [KAMKEEL/CustomNPC-Plus#80]
**Issue description:**

To reproduce...
1. ...simply place an NPC, goto Advanced->Night, enable Editing Mode.
2. Then load day data, goto Apperience and set "Visible" to false. *(it doesn't work also with others, but hey, this is simple to describe)*
3. Then load the night data, goto Apperience and ensure "Visible" is true.
4. Disable the Editing Mode and close the NPC editor window.
5. Now toogle between day and night using `/time set night` and `/time set day`. You see, the NPC wont show or hide itself.
6. Now interact with that NPC by right-clicking it with an empty hand. You see, it now starts to update. *(Side node: In my prod world I changed the position, but it doesn't update the position and animation isntantly, it needs another few seconds and/or interactions.)*

**If crashed, link to crash report (use a site such as pastebin):**
*None.*

**Versions:**
* Minecraft: 1.7.10
* Forge: 10.13.4.1614
* CustomNPC+: 1.7.0

**Can it be reproduced with *just* CustomNPC+? If not, list the other mods *required* to reproduce the issue.**
*No other mods needed to reproduce the issue.*

**Comment by KAMKEEL** [comment]
So you're saying the information on the NPC, like its "NAME", "VISIBILITY" doesn't update until you wand the NPC during Day/Night cycles?

**Comment by Pilzinsel64** [comment]
@KAMKEEL Yes, I need to interact with it once. It doesn't update its information automatically on changing day to night or night to day. I can show in a quick video, if you want.

**Comment by Pilzinsel64** [comment]
@KAMKEEL Sure, here both videos:

https://cloud.pilzinsel64.de/s/tx7tEGcASpMHQwc
--> Shows problem and the configs on the NPCs. You can see that also interacting with the NPC brings the wrong action once. The even mysterical for me is, that even this inconstistence is incosistent: Sometimes it works, but most of the time it don't.

https://cloud.pilzinsel64.de/s/ALSPocZDHPJWKLA
--> Shows the behavior on re-loign

**Comment by KAMKEEL** [comment]
It seems like a client data issue. Especially for the visible/invisible NPC. As for the second NPC, it also may be a client update issue. I'll take a look. It's not necessarily a `Night` issue. It may just be the information from the Server/World not forcing the client to update.

**Comment by Pilzinsel64** [comment]
Huhu @KAMKEEL, I've tested v1.8.3 within my instance and re-checked this bug. It seems to be mostly fixed now. It is not needed to inteact with the NPC once to refresh it as the most data refreshs as soon as the time changes.

Well, at least if the "home" of an NPC is the same. As soon as the "Movement" in "AI" tab is different and you set another "home" at night (e.g. using `/kamkeel npc <name> home <x> <y> <z>`) then it wont update directly for both (new home and new movement).

**Example:**
To force the night data (sit on the chair in the house) I even need to use the NPC wand once to force-respawn it, otherwise it wont update and just stay at its home of the day data.
Then, when swiching to day data (walk randomly outside of the house), I just need to interact with the NPC once to enable the day data for movement and need to wait a little bit (v1.9-sync) for the movement data to load too.

**Comment by KAMKEEL** [comment]
So @Pilzinsel64 can we mark this issue as resolved? As I said, this is likely just client-server desync its much harder to resolve than expected and works _here and there._

**Comment by Pilzinsel64** [comment]
Sure, we can close this for now as the most part does work.
The Movement settings maybe still cause a kind of desync state between client and server, but as you said, it is hard to track down and might depend on the server+client installation.
